src/kernels/raw/finder/jupyterPaths.node.ts

```diff
@@ -126,7 +127,8 @@ export class JupyterPaths {
      * This list comes from the docs here:
      * https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernel-specs
      */
-    @traceDecoratorVerbose('Get Kernelspec root path')
+    @traceDecoratorVerbose('Get KernelSpec root path')
+    @cache(10_000)
     public async getKernelSpecRootPaths(cancelToken?: CancellationToken): Promise<Uri[]> {
         // Paths specified in JUPYTER_PATH are supposed to come first in searching
         const paths = new ResourceSet(await this.getJupyterPathPaths(cancelToken));
```

Comment:
We can cache this, 10 minutes is good enough.
Why 10, simple, why not 10. I felt 1 was too small